### ISSUE TITLE
Script does not run on OpenSuse, seems Debian/Ubuntu specific

### ISSUE THREAD

**Issue by danfos** [PanderMusubi/locale-en-nl#2]
Great to have a way to generate en_NL but unfortunately it does not work for me on OpenSuse and the reason seems to be that "locale-gen" is missing, it seems to be Debian/Ubuntu specific.

Found [here](https://github.com/clearlinux/distribution/issues/2) a work-around in using /etc/locale.conf

```
cp -f en_NL /usr/share/i18n/locales/
localedef -i /usr/share/i18n/locales/en_NL -f UTF-8 en_NL.UTF-8
echo "en_NL.UTF-8" > /etc/locale.conf
```
<reboot>

After the reboot I do not see an errors being reported executing "locale" so that is good.
Things that I see in the boot log are:
```
kwin_x11[2297]: kwin_xkbcommon: XKB: couldn't find a Compose file for locale "en_NL.UTF-8"

tracker-extract[3609]: Could not insert metadata for item "file:///home/user/Documents/xxxx.pdf": 22.49: invalid UTF-8 character
```

**Comment by mpsijm** [comment]
This is fixed by #3, right? :slightly_smiling_face:

**Comment by PanderMusubi** [comment]
I don't support OpenSUSE myself, but please test it or check with @danfos

**Comment by danfos** [comment]
Thanks for merging the patch, issue can be closed I think.

For people running OpenSuse, if you get, after running this script still:

"kwin_xkbcommon: XKB: couldn't find a Compose file for locale "en_NL.UTF-8"

Open /usr/share/X11/locale/compose.dir and search for entries for entries with "en_BE". For every entry with "en_BE" add an entry with "en_NL".

**Comment by PanderMusubi** [comment]
@danfos can you make a PR for the README.md with what you wrote? Thanks.

**Comment by danfos** [comment]
Yes, I can make a PR, I assume for what I wrote about compose.dir, i.e. the Compose file?

Had a read in https://www.x.org/archive/X11R7.6/doc/man/man5/Compose.5.xhtml based on that:

/usr/share/X11/locale/compose.dir  --- File listing the compose file path to use for each locale.
/usr/share/X11/locale/<localemapping>/Compose ---System default compose file for the locale, mapped via compose.dir.

On OpenSuse compose.dir has 4 sections (--- section X --- added) and just filtering on nl_NL you get:

```
$ grep nl_NL /usr/share/X11/locale/compose.dir
--- section 1 ---
iso8859-1/Compose               nl_NL.ISO8859-1
iso8859-15/Compose              nl_NL.ISO8859-15
--- section 2 ---
en_US.UTF-8/Compose             nl_NL.UTF-8
--- section 3 --
iso8859-1/Compose:              nl_NL.ISO8859-1
iso8859-15/Compose:             nl_NL.ISO8859-15
--- section 4 ---
en_US.UTF-8/Compose:            nl_NL.UTF-8
```

What I did is adding en_NL variants of this.

I am wondering if other distributions also have compose.dir like this?

**Comment by danfos** [comment]
OpenSuse functionality was broken by [this commit](https://github.com/PanderMusubi/locale-en-nl/commit/261159e5c0124b7d8c8d19cf73815c791989e8ae) and what I wrote above on compose.dir has been implemented (partly) by [this commit](https://github.com/PanderMusubi/locale-en-nl/commit/befdae7f01e0f6a797b761649eca9ee8d08ae37d).

I had to do a new install so I did update the script to fix OpenSuse support and update the compose.dit/locale.dir handling, see #12.
